client/components/SignUpForm.js

```diff
@@ -1,11 +1,95 @@
-import React from 'react'
+import React from 'react';
+import {connect} from 'react-redux';
+import {signUpUser, getCurrentUser} from '../actions';
+import axios from 'axios';
+
+class SignUpForm extends React.Component {
+  constructor(props){
+    super(props);
+    this.state = {
+      firstName: '',
+      lastName: '',
+      email: '',
+      password: '',
+      cPassword: '',
+      streetAddress:'',
+      city: '',
+      zip:'',
+      state:'',
+    }
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+handleChange(ev){
+  const {name, value} = ev.target;
+  this.setState({[name]: value})
+}
+
+handleSubmit(ev){
+  ev.preventDefault();
+  const sObj = {}
+  for (let key in this.state){
+    if (key !== 'cPassword'){
+      sObj[key] = this.state[key]
+    }
+  }
+  axios.post('/api/users/signup', sObj)

```

Comment:
Should this axios call also made an action that sets user the redux store's state?

Reply:
I'm just setting the userId with the signUp call. Then after that, I call getUser and populate the store that way. I know it's not the most efficient way, but I felt getting a simple answer in place and getting the code out would be best.